src/DynamoDbModel.php

```diff
@@ -86,11 +86,38 @@ protected function setupDynamoDb()
 
     public function save(array $options = [])
     {
-        if (!$this->getKey()) {
-            $this->fireModelEvent('creating');
+        $create = !$this->exists;
+        if ($create) {
+            if ($this->fireModelEvent('creating')  === false) {
+                return false;
+            }
+        } else {
+            if ($this->fireModelEvent('updating') === false) {
+                return false;
+            }
+        }
+
+        if ($this->fireModelEvent('saving') === false) {
+            return false;
         }
 
-        return $this->newQuery()->save();
+        if ($this->usesTimestamps()) {
+            $this->updateTimestamps();
```

Comment:
This is not explicitly implemented for DynamoDb. Either we implement it first then put it here or I'd leave it out since it's breaking the tests.